app/code/Magento/Backup/Model/BackupFactory.php

```diff
@@ -41,20 +41,18 @@ public function create($timestamp, $type)
         $backupId = $timestamp . '_' . $type;
         $fsCollection = $this->_objectManager->get(\Magento\Backup\Model\Fs\Collection::class);
         $backupInstance = $this->_objectManager->get(\Magento\Backup\Model\Backup::class);
+
         foreach ($fsCollection as $backup) {
-            if ($backup->getId() == $backupId) {
-                $backupInstance->setType(
-                    $backup->getType()
-                )->setTime(
-                    $backup->getTime()
-                )->setName(
-                    $backup->getName()
-                )->setPath(
-                    $backup->getPath()
-                );
+            if (strpos($backup->getId(), $backupId) !== false) {
```

Comment:
Hi,

The reason for using strpos was that the id returned by the collection has the name appended to the time stamp and type.
$backup->getId() ==> 1503256588_nomediasecond (notice no underscore between nomedia and 'second' which is the name of the backup)

Could be a bug in the _generateRow method of Magento\Backup\Model\Fs\Collection.

The files created in the backup folder have the name convention of {{time_stamp}}_{{type}}_{{name}}.tgz

The id returned by the collection does not have the second underscore.

In the mass delete controller action only the time stamp and type is used to identify the backup.
Magento\Backup\Controller\Adminhtml\Index\MassDelete:39
```
 list($time, $type) = explode('_', $id);
 $backupModel = $this->_backupModelFactory->create($time, $type)->deleteFile();
```

I updated the comparison to check for the type and time stamp only like in the mass delete action.

Hope that is ok.